test/extended/compatibility.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+#
+# Runs extended compatibility tests with a previous version
+source "$(dirname "${BASH_SOURCE}")/../../hack/lib/init.sh"
+source "${OS_ROOT}/test/extended/setup.sh"
+
+# Previous version to test against
+PREVIOUS_VERSION="v1.2.0"
+
+if [[ -n "${PREVIOUS_API:-}" ]]; then
```

Comment:
You can achieve this defaulting with Bash:
```sh
export API_SERVER_VERSION="${PREVIOUS_API:-"${PREVIOUS_VERSION}"}"
```

Although, I still think we'd gain readability by explicitly passing these to `os::test::extended::setup` as arguments